deployment/terraform/engine.go

```diff
@@ -97,22 +97,26 @@ func (t *Terraform) runCommand(dst io.Writer, args ...string) error {
 }
 
 func checkTerraformVersion() error {
-	out, err := exec.Command("terraform", "version").Output()
+	versionInfoJSON, err := exec.Command("terraform", "version", "-json").Output()
 	if err != nil {
 		return fmt.Errorf("could not run %q command: %w", "terraform version", err)
 	}
 
-	re := regexp.MustCompile(`v\d+.?\d+`)
-	if !re.Match(out) {
-		return fmt.Errorf("could not parse terraform command output: %s", out)
+	var versionInfo struct {
+		Version string `json:"terraform_version"`
 	}
 
-	version, err := strconv.ParseFloat(string(re.Find(out)[1:]), 64)
-	if err != nil {
+	if err := json.Unmarshal(versionInfoJSON, &versionInfo); err != nil {
 		return fmt.Errorf("could not parse terraform command output: %w", err)
 	}
-	if version != supportedVersion {
-		return fmt.Errorf("Terraform version %.2f is required, you have %.2f", supportedVersion, version)
+
+	installedVersion, err := semver.Parse(versionInfo.Version)
+	if err != nil {
+		return fmt.Errorf("could not parse installed version: %w", err)
+	}
+
+	if installedVersion.LT(minimumRequiredVersion) {
+		return fmt.Errorf("installed version %q is lower than supported version %q", installedVersion.String(), minimumRequiredVersion.String())
```

Comment:
Wondering if we should also check for major versions as well. Potential `2.x`+ versions can be shipped and the promise is only for `1.x` versions.

Reply:
Yes, that's true. I was thinking about revisiting this when 2.0 was about to be released, but it does not hurt to check it here now. I'll do it, thanks!